### PR TITLE
Update dependencies and make tests less racy

### PR DESCRIPTION
**Pull request by leodasvacas**
This bumps the minor version of our dependencies (except for `serde-json` with which we have some issue). Also now that deployments go through the database, and also due to the tokio update, some tests became very racy both locally and on CI, so I tried to improve that situation by adding some delays and not testing the order of events.

I expected diesel to have made a release with the fix for the warnings, but they haven't made a release in while. So lets just ignore those warnings for now.

The test `emits_an_api_schema_after_one_schema_is_added` was also being flaky on CI, not sure why. It tested `SchemaEvent` which will probably not exist after #534 so I removed the test.

### CODE REVIEW

**Comment by leodasvacas** [comment]
The latest version of web3 (0.5) has does not support Rust versions older than 1.30 (current stable), so if we want to update it we must also bump our minimum Rust, which I did. I also took advantage of the new macro imports feature to do some cleanup.

**Comment by leodasvacas** [comment]
This caused some issue that's making the node fail to fetch blocks from ethereum (it seems something doesn't accept a leading `0x` anymore). I'll investigate.

File: graph/Cargo.toml [hunk 1]
```diff
@@ -14,13 +14,20 @@ failure = "0.1.2"
...
+# Pin to serde_json 1.0.25. Newer releases use ryu instead of dtoa for floating
+# point (de)serialization, which caused test failures. What may be happening
+# here is that we write entities to the store with f32 values and then what
+# comes back when deserializing is not f32 and so serde_json cannot match float
+# values to our Value::Float(f32) variant. Would have to investigate further how
+# we can fix this.
+serde_json = { version = "=1.0.25", features = ["arbitrary_precision"] }
 serde_yaml = "0.7"
-slog = { version = "2.2.3", features = ["release_max_level_debug"] }
+slog = { version = "2.2.3", features = ["release_max_level_debug", "max_level_trace"] }
```
Comment (timmclean) [inline graph/Cargo.toml]: Nice, so this means we can get trace level logging now if we pass in `GRAPH_LOG=trace`? Is it overwhelming or useful? `trace!()` calls are still compiled out for release builds, right?

Comment (leodasvacas) [reply graph/Cargo.toml]: Yes, `GRAPH_LOG=trace` now works. `trace` is meant to be overwhelming but also useful when all else fails, that said `Run with retry` is particularly spammy. Yes, this is only for debug builds, they're still compiled out of release builds.

File: store/postgres/src/store.rs [hunk 2]
```diff
@@ -861,7 +863,7 @@ impl ChainStore for Store {
         .map(|hashes: Vec<String>| {
             hashes
                 .into_iter()
-                .map(|h| h.parse())
+                .map(|h| h.trim_left_matches("0x").parse())
```
[the remark this answers is not in the document]
Comment (leodasvacas) [reply store/postgres/src/store.rs]: You mean if this is run against an existing database? I suppose this is one of those cases were we ask users to please reset their DBs.

[the remark this answers is not in the document]
Comment (timmclean) [reply store/postgres/src/store.rs]: dropdb is fine. Previously all hashes were stored without `0x`, so something changed that makes this needed now?

[the remark this answers is not in the document]
Comment (leodasvacas) [reply store/postgres/src/store.rs]: @timmclean Are you sure that's the case? If I try reverting the change to this line, I start getting "Invalid character 'x' at position 1" on this new version of web3.

[the remark this answers is not in the document]
Comment (timmclean) [reply store/postgres/src/store.rs]: Hmm sounds like the web3 upgrade did it

**Comment by leodasvacas** [comment]
So we've gotten the story straight now: We were using `{:#x}` (leading `0x`) when we meant `{:x}` (no `0x`), but that was ok because parity did not respect `#x`. With this upgrade they fixed that, which broke us, so the correct fix is to no longer use `#x`. Thanks @timmclean for bringing attention to this.

**Comment by timmclean** [comment]
Looks good! I'm excited about the Diesel warnings being suppressed

**Pull request merged**